edb/edgeql/compiler/viewgen.py

```diff
@@ -1313,7 +1313,7 @@ def _compile_view_shapes_in_set(
             element = setgen.extend_path(
                 path_tip,
                 ptr,
-                unnest_fence=True,
+                unnest_fence=False,
```

Comment:
Hm.  Curious why is this needed and why doesn't this trip `test_edgeql_ir_scope_tree_bad_03`?

Reply:
Having an `unnest_fence` there was causing spurious failures in cases like `test_edgeql_scope_ref_outer_03` and the new `test_edgeql_scope_ref_outer_07`, where we define an alias that refers to an outer scope and then use the alias in a semi-joiny way.

It doesn't trip `test_edgeql_ir_scope_tree_bad_03` because we use an unnest fence when normalizing ptr exprs

Reply:
Actually I should just drop the `unnest_fence` argument altogether if I'm doing this, too

Reply:
Oh, that's a nice cleanup!